### PR TITLE
Use scipy for mad

### PR DESCRIPTION
**Pull request by Dradeliomecus**
Use the [scipy definition](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.median_abs_deviation.html#scipy.stats.median_abs_deviation) for the median absolute deviation rather than hard coding it.

### CODE REVIEW

**Comment by Dradeliomecus** [comment]
The test failed because the calculation of the noise level is slightly different (I believe a bit more precise with scipy).
Some values for the checks are hard coded, this is a problem with the test and not the code.

File: spikeinterface/toolkit/qualitymetrics/tests/test_pca_metrics.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def test_calculate_snrs(simulated_data):
 
     we = setup_dataset(simulated_data, score_detection=0.5)
     snr = compute_snrs(we)
-    assert snr == {0: 12.921033123982426, 1: 12.992882660926739, 2: 12.992882660926739}
+    assert snr == {0: 12.92083677401068, 1: 12.992685219118576, 2: 12.992685219118576}
```
Comment (alejoe91) [inline spikeinterface/toolkit/qualitymetrics/tests/test_pca_metrics.py:141]:
```suggestion
    assert np.allclose(snr.values(), np.array([12.92, 12.99, 12.99]), atol=0.01}
```

**Comment by alejoe91** [comment]
So I see now that all values ar hardcoded...is it the intended behavior? @MarineChap

**Comment by Dradeliomecus** [comment]
I don't think hardcoding the values make much sense.
It means that if we add a new method that gives a slightly different result (like it is the case here), the test will fail.

The idea should be that the metrics converges to a certain value given a certain tolerance?

**Comment by samuelgarcia** [comment]
Cool.
Ok for me.

**Comment by MarineChap** [comment]
Hello, hard coded values are coming from spikemetric tests. Unit test should fail when you are changing even slightly your implementation, so seems to have work correctly for me. 
But if you want to add a small range why not, as you want. Through be careful, more your range is wide, less you can ensure there is no regression in your future code.

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**